apps/dh/app-dh/src/main.ts

```diff
@@ -55,7 +56,24 @@ if (environment.authDisabled) {
   }
 }
 
-Promise.all([loadDhApiEnvironment(), loadDhB2CEnvironment(), loadDhAppEnvironment()])
+let serviceWorkerPromise: Promise<void> | null = null;
+
+if (environment.mocked) {
+  // Dynamically import the MSW setup to avoid loading it in production
+  Promise.all([
+    import('@energinet-datahub/gf/util-msw'),
+    import('@energinet-datahub/dh/shared/data-access-mocks'),
+  ]).then(([{ setupServiceWorker }, { mocks }]) => {
+    serviceWorkerPromise = setupServiceWorker(dhLocalApiEnvironment.apiBase, mocks);
+  });
+}
+
+Promise.all([
+  loadDhApiEnvironment(),
+  loadDhB2CEnvironment(),
+  loadDhAppEnvironment(),
+  serviceWorkerPromise,
+])
```

Comment:
issue: It feels like there's a race condition here. While the `Promise.all` inside the `if` executes, the code will continue to the next `Promise.all`. If the first `Promise.all` completes by that time, `setupServiceWorker` will execute and the mocks will be set up correctly. Otherwise the setup will run after the app has bootstrapped which can make the app unstable.

Reply:
Could be fixed now :-)